async.c

```diff
@@ -57,7 +58,7 @@ static unsigned int callbackHash(const void *key) {
 
 static void *callbackValDup(void *privdata, const void *src) {
     ((void) privdata);
-    redisCallback *dup = malloc(sizeof(*dup));
+    redisCallback *dup = hi_malloc(sizeof(*dup));
```

Comment:
Can still result in a SEGFAULT,
If hi_malloc returns NULL -> next call to memcpy will SEGFAULT

Reply:
Just waking up here but how can hi_malloc return NULL if `HIREDIS_OOM_HANDLER` is called? (eg. `_exit`?)

Reply:
@lamby yeah, I forgot to redo the comment, talked about it already with @michael-grunder 
either way personally I'm against doing `abort()` if we can handle it, for example when doing callback-allocation, we can return error instead of abort, remember that this is a 3rd-party library for most users, and crashing someone's application due to internal errors is not the way to do things IMHO